lib/TerminalDisplay.cpp

```diff
@@ -729,12 +756,15 @@ void TerminalDisplay::drawCharacters(QPainter& painter,
     // This was discussed in: http://lists.kde.org/?t=120552223600002&r=1&w=2
         if (_bidiEnabled)
             painter.drawText(rect,0,text);
-        else
+        else {
 #if QT_VERSION >= 0x040800
-            painter.drawText(rect, Qt::AlignBottom, LTR_OVERRIDE_CHAR + text);
+            QRect drawRect(rect.topLeft(), rect.size());
```

Comment:
Haven't tested myself, but could you fix this issue for all versions by using drawRect on both sides of the #if here?